src/Umbraco.Web.BackOffice/Filters/FileUploadCleanupFilterAttribute.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Umbraco.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Web.Models.ContentEditing;
+
+namespace Umbraco.Web.WebApi.Filters
+{
+    /// <summary>
+    ///     Checks if the parameter is IHaveUploadedFiles and then deletes any temporary saved files from file uploads
+    ///     associated with the request
+    /// </summary>
+    internal sealed class FileUploadCleanupFilterAttribute : TypeFilterAttribute
+    {
+        /// <summary>
+        ///     Constructor specifies if the filter should analyze the incoming or outgoing model
+        /// </summary>
+        /// <param name="incomingModel"></param>
+        public FileUploadCleanupFilterAttribute( bool incomingModel = true) : base(typeof(FileUploadCleanupFilter))
+        {
+
+            Arguments = new object[]
+            {
+                incomingModel
+            };
+        }
+
+        // We need to use IAsyncActionFilter even that we dont have any async because we need access to
+        // context.ActionArguments, and this is only available on ActionExecutingContext and not on
+        // ActionExecutedContext
+
+        private class FileUploadCleanupFilter : IAsyncActionFilter
+        {
+            private readonly ILogger _logger;
+            private readonly bool _incomingModel;
+
+            public FileUploadCleanupFilter(ILogger logger, bool incomingModel)
+            {
+                _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+                _incomingModel = incomingModel;
+            }
+
+
+            public Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+            {
+                next(); // We only to do stuff after the action is executed
+
+                var tempFolders = new List<string>();
+
+                if (_incomingModel)
+                {
+                    if (context.ActionArguments.Any())
+                    {
+
+                        if (context.ActionArguments.First().Value is IHaveUploadedFiles contentItem)
+                        {
+                            //cleanup any files associated
+                            foreach (var f in contentItem.UploadedFiles)
+                            {
+                                //track all temp folders so we can remove old files afterwards
+                                var dir = Path.GetDirectoryName(f.TempFilePath);
+                                if (tempFolders.Contains(dir) == false)
```

Comment:
I just copied the file from the old Umbraco.Web project. I'm pretty sure it is just coding preferences.

The only thing that can be performance optimised is null checks. `x is null` vs `x == null`.  The `is` operator is way faster because it does not check operator overload - But this also makes them a bit different.